RFunction.R

```diff
@@ -1,149 +1,158 @@
-library('move')
-library('sp')
-library('sf')
-library('ggplot2')
-#library('adehabitatLT')
-library('geosphere')
-
-# plans for changes: 
+library("move")
+library("sp")
+library("sf")
+library("ggplot2")
+# library('adehabitatLT')
+library("geosphere")
+
+# plans for changes:
 # 1. find crossings for each individual separately, make track into multiline (of each 2-pt segment) to have more control of when the crossing took place and have a correct timestamp and closest location (in time)
 # 2. have legend indicate also colours black and orange
 # 3. add background map (ggmap or leaflet)
 # 4. option to get back fallback roads file (MoveApps feature)
 
 
-rFunction <- function(data,colour_name=NULL,road_files=NULL)
-{
-  Sys.setenv(tz="UTC")
- 
-  #roads <- st_read("GRIP_roads_NASAY2Y/GRIP_roads_NASAY2Y.shp")
-  roads <- st_read(paste0(getAppFilePath("road_files"),"roads.shp"))
-  
-  data_lsl <- lapply(move::split(data), function(x){Lines(list(Line(coordinates(x))),ID=namesIndiv(x))})
-  data_sldf <- SpatialLinesDataFrame(SpatialLines(data_lsl,proj4string = data@proj4string), data=data.frame(idData(data),"trackId"=unique(trackId(data))) ,match.ID = F)
+rFunction <- function(data, colour_name = NULL, road_files = NULL) {
+  Sys.setenv(tz = "UTC")
   
-  #data_ltraj <- as(data,"ltraj") #ltraj trows away projection info, thus dont use it!
-  #data_spdf <- ltraj2sldf(data_ltraj,byid=TRUE)
+  # roads <- st_read("data/raw/Roads_250k_Canvec_WGS84.shp/Roads_250k_Canvec_WGS84.shp")
+  roads <- st_read(paste0(getAppFilePath("road_files"), "roads.shp"))
+
+  data_lsl <- lapply(move::split(data), function(x) {
+    Lines(list(Line(coordinates(x))), ID = namesIndiv(x))
+  })
   
+  data_sldf <- SpatialLinesDataFrame(
+    SpatialLines(data_lsl, proj4string = data@proj4string),
+    data = data.frame(idData(data), "trackId" = unique(trackId(data))),
+    match.ID = F)
+
+  # data_ltraj <- as(data,"ltraj") #ltraj trows away projection info, thus dont use it!
+  # data_spdf <- ltraj2sldf(data_ltraj,byid=TRUE)
+
   data_sf <- st_as_sf(data_sldf)
-  #data_sf <- st_as_sf(data)
-  #st_crs(data_sf) <- st_crs(roads)
-  roads <- st_transform(roads,st_crs(data_sf))
-  
+  st_geometry(data_sf) <- "geometry"
```

Comment:
The only actual, non-style change

Reply:
@andreakoelzsch does this seem reasonable to you?

Reply:
Hi @smthfrmn, this App is still in move1, so I would rather not rebuild it, as many dependencies migth have changed... Nilanjan had volunteered last time to update this App to move2. I am not sure how far along he is with that. Please would you get in contact and join forces? Thanks! 
@nilanjanchatterjee  FYI

Reply:
okay per Ashley, we can leave the app as if for now. I'm down to upgrade it to move2 after the workshop. Thanks for reviewing @andreakoelzsch